esti/migrate_test.go

```diff
@@ -253,3 +336,209 @@ func testPostMigrateActions(t *testing.T) {
 	require.NoError(t, err, "failed to list runs")
 	require.Equal(t, len(branchResp.JSON200.Results), 3)
 }
+
+func testPreMigrateAuth(t *testing.T) {
+	ctx, _, repo := setupTest(t)
+
+	// creating a viewer, developer, superuser and admin and verifying their roles and permissions
+	viewerCreds := createUserWithCredentialsInGroup(t, ctx, "testViewer", auth.ViewersGroup)
+	developerCreds := createUserWithCredentialsInGroup(t, ctx, "testDeveloper", auth.DevelopersGroup)
+	superUserCreds := createUserWithCredentialsInGroup(t, ctx, "testSuperUser", auth.SuperUsersGroup)
+	adminCreds := createUserWithCredentialsInGroup(t, ctx, "testAdmin", auth.AdminsGroup)
+
+	verifyUserPermissions(t, ctx, repo, "viewer", viewerCreds, viewerPermissions)
+	verifyUserPermissions(t, ctx, repo, "developer", developerCreds, developerPermissions)
+	verifyUserPermissions(t, ctx, repo, "superUser", superUserCreds, superUserPermissions)
+	verifyUserPermissions(t, ctx, repo, "admin", adminCreds, adminPermissions)
+
+	// creating a group with permissions, adding a user and verify permissions
+	// the created group (and so, the user) has permissions to only list repositories, and so
+	// is expected to succeed with that but to fail reading the repository. This is done to
+	// later verify that a custom created groups and policies are migrated correctly
+	respCreateGroup, err := client.CreateGroupWithResponse(ctx, api.CreateGroupJSONRequestBody{
+		Id: authCustomGroupName,
+	})
+	require.NoError(t, err, "Admin failed while creating group")
+	require.Equal(t, http.StatusCreated, respCreateGroup.StatusCode(), "Admin unexpectedly failed to create group")
+
+	pid := "ListReposPolicy"
+	respCreatePolicy, err := client.CreatePolicyWithResponse(ctx, api.CreatePolicyJSONRequestBody{
+		CreationDate: api.Int64Ptr(time.Now().Unix()),
+		Id:           pid,
+		Statement: []api.Statement{
+			{
+				Action:   []string{"fs:ListRepositories"},
+				Effect:   "allow",
+				Resource: "*",
+			},
+		},
+	})
+	require.NoError(t, err, "Admin failed while creating policy")
+	require.Equal(t, http.StatusCreated, respCreatePolicy.StatusCode(), "Admin unexpectedly failed to create policy")
+
+	respAddPolicy, err := client.AttachPolicyToGroupWithResponse(ctx, authCustomGroupName, pid)
+	require.NoError(t, err, "Admin failed while adding policy to group")
+	require.Equal(t, http.StatusCreated, respAddPolicy.StatusCode(), "Admin unexpectedly failed to add policy to group")
+
+	uid := "test-user"
+	customCreds := createUserWithCredentialsInGroup(t, ctx, uid, authCustomGroupName)
+	verifyUserPermissions(t, ctx, repo, "customUser", customCreds, customPermissions)
+
+	// Hardening relevant test data for post-migrate
+	state.Auth.Repo = repo
+	state.Auth.ViewerUser.Username = "testViewer"
```

Comment:
You don't have to pass const data to the state, you can simply use the same values from the post test

Reply:
Moreover, the `username` is not really use after the user creation (in `PreTest`) as the users are identified by the credentials. Removed the field all together